### PR TITLE
First commit new cookbook - PII tutorial

### PR DESCRIPTION
**Pull request by vincenzopecorella**
Added new cookbook containing a new tutorial about PII instrument

### CODE REVIEW

File: docs/cookbook/pii_tutorial.md [hunk 1]
```diff
@@ -0,0 +1,207 @@
+In this tutorial, we’ll explore practical, real-world applications of Eunomia, focusing specifically on Personal Identifiable Information (PII). We’ll discuss how Eunomia’s capabilities and built-in tools for PII management enable new and innovative use cases.
+
+## What You’ll Discover Today
```
Comment (tommitt) [inline docs/cookbook/pii_tutorial.md:3]: This title looks odd, it should rather be "what you'll discover in this tutorial" or similar

File: docs/cookbook/pii_tutorial.md [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+### The Nightmare of PII Exposure
+
+Tickets in your system contain a wide range of Personal Identifiable Information (PII): names, roles, passwords, and other sensitive data. How can you ensure that sensitive information is never retrieved, avoiding catastrophic data leaks within your company?
+
+### Enter Eunomia
+
+Eunomia’s advanced PII management instruments ensure that sensitive data—such as names, emails, and phone numbers—are effectively safeguarded.
+
+#### Processing Methods
```
Comment (tommitt) [inline docs/cookbook/pii_tutorial.md:35]: This part is a bit hard to understand without further clarification. I think it's important to mention that eunomia can be inserted in different parts of the data pipeline and the choice goes to the developer

File: docs/cookbook/pii_tutorial.md [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+### Developing the Solution with Eunomia
+
+#### Step 1: Setup
+
+To begin, ensure that Eunomia is installed by following the [installation guide](../get_started/installation.md). Then, import the necessary components in your script:
+
+```python
+from eunomia.orchestra import Orchestra
+from eunomia.instruments import PiiInstrument
+```
```
Comment (tommitt) [inline docs/cookbook/pii_tutorial.md:71]: Can we add a reference to Orchestra and PiiInstrument api docs?

File: docs/cookbook/pii_tutorial.md [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+    # Clean the PII from the current ticket
+    ticket_text_cleaned = eunomia.run(ticket_text)
+
+    # Append the sanitized ticket to the cleaned list
+    tickets_cleaned.append(ticket_text_cleaned)
+
+for i, cleaned_ticket in enumerate(tickets_cleaned, start=1):
+    print(f"Cleaned Ticket {i}:\n{cleaned_ticket}\n")
+```
+
```
Comment (tommitt) [inline docs/cookbook/pii_tutorial.md:122]: Let's add the `print` outputs as comments at the bottom of the code block

File: docs/cookbook/pii_tutorial.md [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+2. **Document Ownership**: Every document is linked to the unique ID of its owner.
+3. **Global PII Masking**: Specific types of PII are consistently masked, regardless of the viewer.
+4. **Owner-Specific PII Access**: Owners can view their own PII while it remains masked for others.
+
+#### Step 1: Setup
+
+```python
+from eunomia.orchestra import Orchestra
+from eunomia.instruments import PiiInstrument, IdbacInstrument
+```
```
Comment (tommitt) [inline docs/cookbook/pii_tutorial.md:143]: Can we add a reference to IdbacInstrument api doc?

File: docs/cookbook/pii_tutorial.md [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+        if ticket["doc_id"] == doc_id:
+            processed_text = eunomia.run(ticket["text"], user_id=user_id, doc_id=doc_id)
+
+            # Output the results
+            access_status = "Matching IDs" if user_id == doc_id else "Non-Matching IDs"
+            print(f"Access Status: {access_status}")
+            print(f"User ID: {user_id}, Document ID: {doc_id}")
+            print(f"Processed Ticket:\n{processed_text}\n")
+
+```
```
Comment (tommitt) [inline docs/cookbook/pii_tutorial.md:203]: In general I think cookbook's code blocks should contain outputs as well. Can we add the output of `print` functions?

**Review by tommitt — CHANGES_REQUESTED** [review]
Minor improvements to cookbook example

**Review by tommitt — APPROVED** [review]

File: mkdocs.yml [hunk 7]
```diff
@@ -77,6 +80,7 @@ nav:
       - IDBAC Instrument: api/instruments/idbac_instrument.md
   - Cookbook:
     - cookbook/index.md
+    - Managing Personally Identifiable Information: cookbook/pii_tutorial.md
```
Comment (tommitt) [inline mkdocs.yml:83]: I would give it a more use case-specific name as "PII in an LLM-powered IT Service Desk" as "Managing PII" is a bit too generic, wdyt @vincenzopecorella?

File: docs/cookbook/pii_tutorial.md [hunk 8]
```diff
@@ -0,0 +1,260 @@
+In this tutorial, we’ll explore practical, real-world applications of Eunomia, focusing specifically on Personal Identifiable Information (PII). We’ll discuss how Eunomia’s capabilities and built-in tools for PII management enable new and innovative use cases.
+
+## What You’ll Discover in this tutorial\
```
Comment (tommitt) [inline docs/cookbook/pii_tutorial.md:3]: typo "\" here, and would put uppercase Tutorial

File: docs/cookbook/pii_tutorial.md [hunk 9]
```diff
@@ -0,0 +1,260 @@
...
+```python
+from eunomia.orchestra import Orchestra
+from eunomia.instruments import PiiInstrument, IdbacInstrument
+```
+
+You can find out more about the components used at:
+
+- [Orchestra docs](../api/orchestra.md)
+- [PiiInstrument docs](../api/instruments/pii_instrument.md)
+- [IdbacInstrument docs](../api/instruments/idbac_instrument.md)
```
Comment (tommitt) [inline docs/cookbook/pii_tutorial.md:172]: Bullet points take a lot of space here, I would rather list them in the same paragraph

File: docs/cookbook/pii_tutorial.md [hunk 10]
```diff
@@ -0,0 +1,260 @@
...
+To begin, ensure that Eunomia is installed by following the [installation guide](../get_started/installation.md). Then, import the necessary components in your script:
+
+```python
+from eunomia.orchestra import Orchestra
+from eunomia.instruments import PiiInstrument
+```
+You can find out more about the components used at:
+
+- [Orchestra docs](../api/orchestra.md)
+- [PiiInstrument docs](../api/instruments/pii_instrument.md)
```
Comment (tommitt) [inline docs/cookbook/pii_tutorial.md:77]: Bullet points take a lot of space here, I would rather list them in the same paragraph

File: docs/cookbook/index.md [hunk 11]
```diff
@@ -2,4 +2,17 @@
...
 ---
 
-Work in progress.
+This section contains practical examples and use cases to help you effectively utilize Eunomia. Whether you're a beginner or an advanced user, you'll find step-by-step solutions and ideas to enhance your workflow.
+
+## Tutorials
+
+| Tutorial                                     | Jump to                                        |
+| -------------------------------------------- | ---------------------------------------------- |
+| Managing Personally Identifiable Information | [:material-arrow-right: Page](pii_tutorial.md) |
```
Comment (tommitt) [inline docs/cookbook/index.md:11]: Change name here as well in case

**Review by tommitt — CHANGES_REQUESTED** [review]
Final tweaks for PII cookbook doc

File: mkdocs.yml (hunk 7 above)
Comment (vincenzopecorella) [reply mkdocs.yml:83]: Yes agree!

**Review by tommitt — APPROVED** [review]

**Review by tommitt — APPROVED** [review]

**Pull request merged**